pkg/app/piped/planner/kubernetes/kubernetes.go

```diff
@@ -480,3 +492,45 @@ func determineVersion(manifests []provider.Manifest) (string, error) {
 
 	return b.String(), nil
 }
+
+// determineVersions decide artifact versions of an application.
+// For example, in case some containers are used in one application, those image version are retured
+func determineVersions(manifests []provider.Manifest) ([]*model.ArtifactVersion, error) {
+	versions := []*model.ArtifactVersion{}
+
+	for _, m := range manifests {
+		if !m.Key.IsDeployment() {
+			continue
+		}
+		data, err := m.MarshalJSON()
+		if err != nil {
+			return nil, err
+		}
+		var d resource.Deployment
+		if err := json.Unmarshal(data, &d); err != nil {
+			return nil, err
+		}
+
+		containers := d.Spec.Template.Spec.Containers
+		for _, c := range containers {
+			image := parseContainerImage(c.Image)
+			versions = append(versions, &model.ArtifactVersion{
+				Kind:    model.ArtifactVersion_CONTAINER_IMAGE,
+				Version: image.tag,
+				Name:    image.name,
+				Url:     c.Image,
+			})
+		}
+	}
```

Comment:
Same container images could be used in multiple manifests so I think we have to remove the duplicate ones.